File: app/src/main/java/aipova/githubsearch/ReposAdapter.java
package aipova.githubsearch;


import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import aipova.githubsearch.dao.RepoModel;

/**
 * Created by aipova on 07.02.17.
 */

public class ReposAdapter extends RecyclerView.Adapter<ReposAdapter.RepoViewHolder>{

    List<RepoModel> repoModelList;

    public void clear() {
        repoModelList.clear();
    }

    public class RepoViewHolder extends RecyclerView.ViewHolder {
        public TextView title, description;
        public RepoViewHolder(View itemView) {
            super(itemView);
            title = (TextView) itemView.findViewById(R.id.repo_title);
            description = (TextView) itemView.findViewById(R.id.repo_description);
        }
    }

    public ReposAdapter(List<RepoModel> repoModelList) {
        this.repoModelList = repoModelList;
    }

    public List<RepoModel> getRepoModelList() {
        return repoModelList;
    }

    public void setRepoModelList(List<RepoModel> repoModelList) {
        this.repoModelList = repoModelList;
    }

    public void addRepoModelsToList(List<RepoModel> repoModelList) {
        this.repoModelList.addAll(repoModelList);
    }

    @Override
    public RepoViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View repoView = LayoutInflater.from(parent.getContext()).inflate(R.layout.repo_item, parent, false);
        return new RepoViewHolder(repoView);
    }

    @Override
    public void onBindViewHolder(RepoViewHolder holder, int position) {
        RepoModel repoModel = repoModelList.get(position);
        holder.title.setText(repoModel.getTitle());
        holder.description.setText(repoModel.getDescription());
    }

    @Override
    public int getItemCount() {
        return repoModelList != null ? repoModelList.size() : 0;
    }


}

File: app/src/main/java/aipova/githubsearch/ReposAdapter.kt
package aipova.githubsearch


import aipova.githubsearch.data.Repo
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

class ReposAdapter(private val repoModelList: MutableList<Repo>) :
    RecyclerView.Adapter<ReposAdapter.RepoViewHolder>() {

    fun clear() {
        repoModelList.clear()
    }

    inner class RepoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var title: TextView = itemView.findViewById<View>(R.id.repo_title) as TextView
        var description: TextView = itemView.findViewById<View>(R.id.repo_description) as TextView
    }

    fun addRepoModelsToList(repoModelList: List<Repo>) {
        this.repoModelList.addAll(repoModelList)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RepoViewHolder {
        val repoView = LayoutInflater.from(parent.context).inflate(R.layout.repo_item, parent, false)
        return RepoViewHolder(repoView)
    }

    override fun onBindViewHolder(holder: RepoViewHolder, position: Int) {
        val repoModel = repoModelList[position]
        holder.title.text = repoModel.name
        holder.description.text = repoModel.description
    }

    override fun getItemCount(): Int {
        return repoModelList.size
    }


}
